pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -495,6 +495,12 @@ func (p *staticPolicy) takeByTopology(availableCPUs cpuset.CPUSet, numCPUs int)
 		}
 		return takeByTopologyNUMADistributed(p.topology, availableCPUs, numCPUs, cpuGroupSize, cpuSortingStrategy)
 	}
+
+	if p.options.AlignByUnCoreCacheOption {
+
+		return takeByTopologyUnCoreCachePacked(p.topology, availableCPUs, numCPUs, cpuSortingStrategy)
+
+	}
```

Comment:
we need to merge this logic in  `takeByTopologyNUMAPacked` passing down the flag where needed, need to think about it

Reply:
Last week we were taking a look at this and we don't think this can be merged because this function is called specifically based on the policy option, we would need to be able to check inside this function right?

Reply:
yes, we can. I see this was fixed in the last upload.